### PR TITLE
multilingual build

### PR DESCRIPTION
**Pull request by gmanchon**
add support for locales for https://github.com/lewagon/data-setup/pull/191

refacto to improve build time from 18s to 5s

the build will pass when https://github.com/lewagon/setup/pull/361 and https://github.com/lewagon/data-setup/pull/191 are merged

### CODE REVIEW

File: build.rb [hunk 1]
```diff
@@ -138,75 +140,71 @@
...
 
-subs = {
-  "WINDOWS.md" => DEFAULT_SUBS,
-  "macOS.md" => DEFAULT_SUBS,
-  "LINUX.md" => DEFAULT_SUBS
-}
+def load_partial(partial, locale)
+  match_data = partial.match(/setup\/(?<partial>[0-9a-z_]+)/)
+  partial = match_data[:partial] if match_data
+  partial = File.join(locale, partial) unless locale.empty?
```
Comment (krokrob) [inline build.rb:163]: Does that mean you keep the 🇬🇧 version if the partial is not localized? I think the build should fail so we got an alert on a missing translation

**Review by krokrob — CHANGES_REQUESTED** [review]
@gmanchon I have a warning about i18n

File: build.rb (hunk 1 above)
Comment (gmanchon) [reply build.rb:163]: @krokrob the function looks for 🇬🇧 content in `content.md` and 🇪🇸 content in `es/content.md`. If some 🇪🇸 content is missing the build will indeed fail when we try to open a file or uri that does not exist.

File: build.rb [hunk 2]
```diff
@@ -138,75 +140,71 @@
...
 
-delimiters = {
-  "WINDOWS.md" => ["\\$WINDOWS_START\n", "\\$WINDOWS_END\n"],
-  "macOS.md" => ["\\$MAC_START\n", "\\$MAC_END\n"],
-  "LINUX.md" => ["\\$LINUX_START\n", "\\$LINUX_END\n"]
-}
+# load partials
+loaded = FILENAMES.map { |filename, (os_name, partials)| partials }.flatten.uniq
+loaded = loaded.map { |partial| LOCALES.map { |locale| [partial, locale]} }.flatten(1)
+loaded = loaded.map { |partial, locale| ["#{partial}.#{locale}", load_partial(partial, locale)] }.to_h
```
[the remark this answers is not in the document]
Comment (gmanchon) [reply build.rb:181]: we speed things up by loading each ressource only once (the ressources are shared between operating systems). this is especially signifiant for ressources from the web setup referenced through uri

File: build.rb [hunk 3]
```diff
@@ -138,75 +140,71 @@
...
-      # remove the OS dependant block delimiters
-      delimiters[os_name].each do |delimiter|
-        content.gsub!(/#{delimiter}/, "")
-      end
-      CONSTANTS.each do |placeholder, value|
-        content.gsub!("<#{placeholder}>", value)
+# write files
+LOCALES.each do |locale|
+  FILENAMES.each do |filename, (os_name, partials)|
+    filename += ".#{locale}" unless locale.empty?
```
[the remark this answers is not in the document]
Comment (gmanchon) [reply build.rb:186]: this generates `README.md` as a destination for 🇬🇧 and `README.es.md` for 🇪🇸, the error will be raised at the moment we try to open a missing file or uri 👍

**Comment by gmanchon** [comment]
@krokrob the force push rebases the changes to the new `data-setup-readme.es.md` branch root

File: build.rb (hunk 1 above)
Comment (gmanchon) [reply build.rb:163]: this is why the build gha of this branch fails since the 🇪🇸 branch of the web setup is not merged

**Comment by gmanchon** [comment]
@krokrob I temporarily removed the support for the `es` locale so that the build passes and we can merge the PR, WDYT ?

**Pull request merged**